### PR TITLE
fix: do not emit any type aliases that shadow ArrayLike.

### PR DESCRIPTION
**Pull request by rkirov**
closes #210

### CODE REVIEW

File: src/main/java/com/google/javascript/clutz/DeclarationGenerator.java [hunk 1]
```diff
@@ -208,6 +207,16 @@ public String produceDts(Depgraph depgraph) {
         declareModule(provide, true);
         continue;
       }
+      // ArrayLike is defined in lib.d.ts, so we skip any type alias that
+      // would shadow it.
+      // Note that clutz expands type aliases used in closure code,
+      // thus this does not result in undefined types.
+      // This case handles goog.provided typedefs.
+      if (isTypedef(symbol.getType()) && symbol.getName().endsWith(".ArrayLike")) {
```
Comment (mprobst) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: Extract a method `isArrayLike(symbol)` or `maybeHandleArrayLike(symbol)`?

Comment (rkirov) [inline src/main/java/com/google/javascript/clutz/DeclarationGenerator.java]: done.

**Comment by rkirov** [comment]
I am assuming you forgot to LGTM it, since you had only one refactoring comment.

**Comment by mprobst** [comment]
Sorry, yes, LGTM.

Rado Kirov <notifications@github.com> schrieb am Do., 7. Jan. 2016, 20:15:

**Pull request merged**